packages/react-router/modules/Redirect.js

```diff
@@ -49,6 +51,24 @@ class Redirect extends React.Component {
       this.perform()
   }
 
+  componentDidUpdate(prevProps) {
+    let prevTo = prevProps.to
+    let nextTo = this.props.to
+
+    prevTo = typeof prevTo === 'string' ? { pathname: prevTo } : prevTo
+    nextTo = typeof nextTo === 'string' ? { pathname: nextTo } : nextTo
+
+    if (locationsAreEqual(prevTo, nextTo))
+      return warning(
```

Comment:
@pshrmn Can you share exactly what you'd expect?  
I think interpolating inside the template string didn't break well.  
Unless you mean to ditch the `%s%s` and do a string concatenation?